src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -135,9 +220,48 @@ protected override void InsertItem(int index, T item)
             CheckReentrancy();
             base.InsertItem(index, item);
 
-            OnCountPropertyChanged();
-            OnIndexerPropertyChanged();
-            OnCollectionChanged(NotifyCollectionChangedAction.Add, item, index);
+            if (!_skipRaisingEvents)
+            {
+                OnCountPropertyChanged();
+                OnIndexerPropertyChanged();
+                OnCollectionChanged(NotifyCollectionChangedAction.Add, item, index);
+            }
+        }
+
+        /// <summary>
+        /// Called by base class Collection&lt;T&gt; when a collection of items is added to list;
+        /// raises a CollectionChanged event to any listeners.
+        /// </summary>
+        protected override void InsertItemsRange(int index, IEnumerable<T> collection)
+        {
+            CheckReentrancy();
+
+            bool ignore = _skipRaisingEvents;
+            if (!ignore)
+            {
+                _skipRaisingEvents = true;
+            }
+
+            try
+            {
+                base.InsertItemsRange(index, collection);
+            }
+            finally
+            {
+                if (!ignore)
+                {
+                    _skipRaisingEvents = false;
+                }
+            }
+
+            if (!_skipRaisingEvents)
+            {
+                List<T> list = collection is List<T> ? (List<T>)collection : new List<T>(collection);
```

Comment:
Similarly to above, this should be something like:

```c#
IList changedItems = collection is IList list ? list : new List<T>(collection);
```

Reply:
This syntax did not work well as mentioned above, I think a good compromise will be the following syntax which I also cited above.

```c#
IList newItems = collection is IList ? (IList)collection : new List<T>(collection);
```

Reply:
This one we were actually able to use your provided syntax. I just pushed theses changes. I'm going to resolve this one. Feel free to open on the latest code base if you find any other issues